### PR TITLE
plat-k3: Add DDR setup in k3 platform

### PR DESCRIPTION
**Pull request by fr0st61te**
This patch introduces DDR setup for possible use of
CFG_CORE_DYN_SHM/dynamic shared memory on k3 platform.

Signed-off-by: Ivan Mikhaylov <ivan.mikhaylov@siemens.com>

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by fr0st61te** [comment]
@manorit2001 could you please look into this patch?

File: core/arch/arm/plat-k3/main.c [hunk 1]
```diff
@@ -37,6 +37,10 @@ register_phys_mem_pgdir(MEM_AREA_IO_SEC, SEC_PROXY_SCFG_BASE,
 			SEC_PROXY_SCFG_SIZE);
 register_phys_mem_pgdir(MEM_AREA_IO_SEC, SEC_PROXY_RT_BASE, SEC_PROXY_RT_SIZE);
 
+#ifdef DRAM0_BASE
+register_ddr(DRAM0_BASE, DRAM0_SIZE);
+#endif
```
Comment (manorit2001) [inline core/arch/arm/plat-k3/main.c:42]: The `#ifdef` guards ain't required I think as the platform_config will always define these values based on the position of the `#define`

**Review by manorit2001 — APPROVED** [review]
`Acked-by: Manorit Chawdhry <m-chawdhry@ti.com>`

**Pull request merged**